src/firehose.rs

```diff
@@ -0,0 +1,3 @@
+//! Amazon Kinesis Firehose
+
```

Comment:
I commented below regarding this, but while I personally don't think having the "Exception" on the end of every enum variant is rust-y, it's not that big of an issue, so it's OK if it's left in.

BUT if you do choose to leave it in, then these `cfg_attr` lines are needed to suppress clippy's warnings.